### ISSUE TITLE
Request: condition `push_preview` on GitHub PR label

### ISSUE THREAD

**Issue by tkf** [JuliaDocs/Documenter.jl#1225]
`push_preview` is a quite nice feature but pushing previews to `gh-pages` for every push to every PR is likely to bloat the repository size (although it's true that I can clean up the history in `gh-pages` anytime).  It would be nice if there is an option to require (e.g.) a label to be set for `push_preview` to work.

**Comment by mortenpi** [comment]
Technical question: is it possible to have GitHub Actions listen to label change events and restart the documentation build somehow (I am thinking about the case where you're applying the label to an existing PR)?

**Comment by tkf** [comment]
Yes, it seems that GitHub Actions has [label event](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#label-event-label).  But I was thinking more like just conditioning the push on the label.  So, the build is still triggered by the push event but `deploydocs` checks if a certain label is set via some GitHub API.

**Comment by tkf** [comment]
It turned out that it's super easy to set it up. You just have to parse the JSON file at `$GITHUB_EVENT_PATH`:

```julia
function should_push_preview(event_path = get(ENV, "GITHUB_EVENT_PATH", nothing))
    event_path === nothing && return false
    event = JSON.parsefile(event_path)
    labels = [x["name"] for x in event["pull_request"]["labels"]]
    return "push_preview" in labels
end

deploydocs(
    ...,
    push_preview = should_push_preview(),
)
```

Example: https://github.com/tkf/MyPlayground.jl/pull/28

Ref: https://developer.github.com/v3/activity/events/types/#pullrequestevent

**Comment by odow** [comment]
JuMP et al. have added a `doc_cleanup` workflow that deletes the preview on close:

https://github.com/jump-dev/MathOptInterface.jl/blob/master/.github/workflows/doc_cleanup.yml

This greatly reduces the size of the repositories.
